leader/file/file.go

```diff
@@ -0,0 +1,59 @@
+package file
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/infrakit/leader"
+)
+
+const (
+	// LeaderFileEnvVar is the environment variable that may be used to customize the plugin leader detection
+	LeaderFileEnvVar = "INFRAKIT_LEADER_FILE"
+)
+
+// DefaultLeaderFile is the file that this detector uses to decide who the leader is.
+// In a mult-host set up, it's assumed that the file system would be share (e.g. NFS mount or S3 FUSE etc.)
+func DefaultLeaderFile() string {
```

Comment:
This is actually better in a `cmd`/ main that uses the library.  Moving to that.